_episodes/21.md

```diff
@@ -0,0 +1,211 @@
+to build production-grade data pipelines. 
+
+When we speak about dbt, we are talking about two slightly different things. dbt
+core is our open source framework that provides the sql compiler and framework 
+to manage your SQL workflow. You can interact with it via a command line 
+interface. We also have dbt Cloud which is a fully managed SaaS product. You can
+use it to handle all of your dbt projects from development to deployment in a 
+single browser based tool.It provides useful features like a full IDE to develop
+and test code, orchestration, logging, and alerting. At the moment, dbt Cloud is
+not available for Presto/Trino users. 
+
+Speaking to the framework, it allows you also to check the quality of results, 
+document the lineage, manage the changes/versions in the SQL scripts and 
+orchestrate the queries, like a CI/CD framework but for your data. dbt is not an
+extract and load tool. The focus is on transforming what is already in your data
+warehouse/data lake. 
+
+Check out these links to learn more:
+* [https://www.getdbt.com/](https://www.getdbt.com/)
+* [https://docs.getdbt.com/docs/introduction](https://docs.getdbt.com/docs/introduction)
+
+### Goals of dbt and how that differs from Trino
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/21/dbt-trino-architecture.png"/><br/>
+Commander Bun Bun is diving deep to find anomalies!
+</p>
+
+Trino is the execution SQL engine and dbt is the framework to manage your SQL 
+statements. dbt won't execute the SQL itself, rather it pushes all of the 
+compute down to the SQL engine. And this SQL engine can be Trino, or an engine 
+included in the data source like the database itself.
+
+### Transformation vs ad-hoc joins
+
+Transformations done by dbt are in general used to clean and prepare data for 
+analytics purposes. It's often used to go from the raw data to a ready-to-use 
+data for reporting and analysis. dbt will create database objects like tables 
+or views to be consumed by business users and analytics tools.
+
+On the other hand, even if Trino can also execute SQL to create tables and 
+views, these SQL queries are not managed but just executed. Trino doesn't have,
+like dbt, all the framework to version, audit, document and orchestrate SQL 
+script and execution. In general Trino is more used to execute SQL SELECT 
+statements generated by users or BI tools to analyze data in an interactive way.
+
+### Cases for why you need both
+
+Both solutions are complementary when you need to access different sources from
+a single SQL query or when you need to run SQL query with good performance on
+object storages (like S3, GCS, ADLS or HDFS).
+
+It's where Trino can complement dbt, as dbt can only access a single data 
+warehouse connection in a SQL query. In dbt there is no way to query multiple 
+storage systems at the same time.
+
+Also, for object storage (data lake), Trino is recognized to have great 
```

Comment:
Trino is recognized for great performance with object storage/data lake processing. With dbt it can transform and prepare data at scale.